src/main/java/com/redhat/devtools/lsp4ij/usages/LSPUsageTypeProvider.java

```diff
@@ -46,18 +47,17 @@ public UsageType getUsageType(final @NotNull PsiElement element) {
     @Override
     public UsageType getUsageType(PsiElement element, UsageTarget @NotNull [] targets) {
         if (element instanceof LSPUsagePsiElement usageElement) {
-            switch (usageElement.getKind()) {
-                case declarations:
-                    return DECLARATIONS_USAGE_TYPE;
-                case definitions:
-                    return DEFINITIONS_USAGE_TYPE;
-                case typeDefinitions:
-                    return TYPE_DEFINITIONS_USAGE_TYPE;
-                case references:
-                    return REFERENCES_USAGE_TYPE;
-                case implementations:
-                    return IMPLEMENTATIONS_USAGE_TYPE;
-            }
+            return switch (usageElement.getKind()) {
+                case declarations -> DECLARATIONS_USAGE_TYPE;
+                case definitions -> DEFINITIONS_USAGE_TYPE;
+                case typeDefinitions -> TYPE_DEFINITIONS_USAGE_TYPE;
+                case references -> REFERENCES_USAGE_TYPE;
+                case implementations -> IMPLEMENTATIONS_USAGE_TYPE;
+            };
+        }
+        // We can also have local usages to semantic token elements which are references

```

Comment:
This ensures that `LSPSemanticTokenPsiReference`s to `LSPSemanticTokenPsiElement`s are properly categorized as reference usages.